### PR TITLE
feat: use trait for Struct -> StructID instead of renaming objects

### PR DESCRIPTION
**Pull request by kjuulh**
This feature should allow both `id` and regular structs/objects/scalars to be passed in to a function. This was initially brought on by: #6933.

Which now allows removing deprecated input

### CODE REVIEW

File: sdk/rust/crates/dagger-codegen/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#![deny(warnings)]
+//#![deny(warnings)]
```
Comment (jedevc) [inline sdk/rust/crates/dagger-codegen/src/lib.rs:1]: :eyes: is this intentional?

**Review by jedevc — APPROVED** [review]
Ooh I really like the idea of being able to pass both IDs / the objects from the SDK-side - that's really neat!

That definitely resolves the issue in #6933.

File: sdk/rust/crates/dagger-codegen/src/lib.rs (hunk 1 above)
Comment (kjuulh) [reply sdk/rust/crates/dagger-codegen/src/lib.rs:1]: Woops nope. I've got another pr with some changes to this and I didn't manage to get that in. I'll get it fixed =D

**Review by jedevc — CHANGES_REQUESTED** [review]
Rebased, and also fixed the warnings issue.

I also tried updating the use of the old `Container` and couldn't:

```diff
-        .container_opts(dagger_sdk::QueryContainerOpts {
-            id: Some(id),
-            platform: None,
-        })
+        .load_container_from_id(id)
```

```
    --> crates/dagger-sdk/tests/mod.rs:78:33
     |
78   |         .load_container_from_id(id)
     |          ---------------------- ^^ the trait `IntoID<ContainerId>` is not implemented for `ContainerId`
     |          |
     |          required by a bound introduced by this call
     |
```

And there we hit the boundaries of my rust knowledge :laughing:

**Comment by kjuulh** [comment]
> Rebased, and also fixed the warnings issue.
> 
> I also tried updating the use of the old `Container` and couldn't:
> 
> ```diff
> -        .container_opts(dagger_sdk::QueryContainerOpts {
> -            id: Some(id),
> -            platform: None,
> -        })
> +        .load_container_from_id(id)
> ```
> 
> ```
>     --> crates/dagger-sdk/tests/mod.rs:78:33
>      |
> 78   |         .load_container_from_id(id)
>      |          ---------------------- ^^ the trait `IntoID<ContainerId>` is not implemented for `ContainerId`
>      |          |
>      |          required by a bound introduced by this call
>      |
> ```
> 
> And there we hit the boundaries of my rust knowledge 😆 It feels like `ContainerID` should definitely implement `IntoID<ContainerID>`.

Yes that was the whole Idea. I'll have a look =D

**Comment by kjuulh** [comment]
It should work now:

```rust
#[tokio::test]
async fn test_can_handle_old_and_new_load() -> eyre::Result<()> {
    let client = dagger_sdk::connect().await?;

    let base_container = client.container().from("alpine:latest");

    client
        .load_container_from_id(base_container.id().await?)
        .with_exec(vec!["echo", "from-id"])
        .sync()
        .await?;

    client
        .load_container_from_id(base_container.clone())
        .with_exec(vec!["echo", "from-id"])
        .sync()
        .await?;

    client
        .container_opts(QueryContainerOpts {
            id: Some(base_container.id().await?),
            platform: None,
        })
        .with_exec(vec!["echo", "from-id-original"])
        .sync()
        .await?;

    Ok(())
}

```

**Review by jedevc — APPROVED** [review]

**Pull request merged**